views/components/routes.jsx

```diff
@@ -14,10 +15,14 @@ export default function getRoutes(store) {
     }
   };
 
+  const resetLogin = () => {
+    store.dispatch(resetAuthError());
+  };
+
   return (
     <Route path="/" component={App}>
       <IndexRoute component={Landing} />
-      <Route path="login" component={Session} />
+      <Route path="login" component={Session} onLeave={resetLogin} />
```

Comment:
onLeave might have a bug probably in react-router-redux lib. Changing this to onEnter works